### ISSUE TITLE
.nylas-mail /.nylas folder size is unworkable! 

### ISSUE THREAD

**Issue by scgough** [nylas/nylas-mail#3351]
Since Installed Nylas Mail last week I've been suffering with lack of space on my MacBook Pro (120GB drive) with free space hovering around the 2-3GB mark. OS is obviously warning me a lot about this lack of space and I'm suffering a performance hit across the board.
 
I've found that the root cause of this is that my hidden `.nylas*` folders total to over 17GB of space on my machine. I'm presuming this is the cache but it seems like it could be an entire copy of my gmail email folders locally at that size. 
For me this is completely unusable and to quite a shocking find. I've had to uninstall the app but surely there should be a way of limiting this local cache (if that is what it is) to only bring down `X days` worth of history...with the rest having to be fetched from online. 

Someone else I was chatting to on nylas' slack claimed their folder was 22GB. 

To sum up - that is a crazy amount of drive space to be hogging. 

...

##### What operating system are you using?
Mac OS Sierra

##### What version of Nylas Mail are you using?
Latest

--

**Bug?**
##### Do you have any third-party plugins installed? If so, which ones?
No

##### Is the issue related to a specific email provider (Gmail, Exchange, etc.)?
No

##### Is the issue reproducible with a particular attachment, message, signature, etc?
Nylas app issue

--

**Feature Request?**
##### Does this feature exist in another mail client or tool you use?
No

**Comment by alberto2000** [comment]
Can anyone tell me where this folder is located under Linux (Arch)? I can seem to remove the emails Nylas has fetched/synced but I want to do a clean reinstall/uninstall

**Comment by alberto2000** [comment]
Totally agree on having an option to limit email cache size. On MacOS I used Newton Mail which if I remember correctly only caches emails in your inbox but not sent, trash or archived emails - quite a viable solution for me as I'd rather wait some seconds for delected emails to pop up rather than having 20+ GB of email data filling my disc.

**Comment by jstejada** [comment]
thanks for bringing this up! we're working on some improvements to reduce disk usage!

**Comment by ronisbr** [comment]
I also have this problem. My gmail account is huge (10 GiB), which I need to download in every computer I have. This is almost prohibitive for me. I am wondering if there is a way to download the messages by request, since there are a lot of labels (folders) which I normally do not need.

**Comment by brandonmp** [comment]
i see your 10 & raise you 5. please fix :/

![image](https://cloud.githubusercontent.com/assets/14139991/25725125/b3c1930a-30ec-11e7-94b2-12e53311b815.png)

**Comment by moxx** [comment]
+1 a must have feature
